concepts/modularization/README.md

```diff
@@ -0,0 +1,172 @@
+
+New images of our own components (eventing-controller, event-publisher-proxy) would require changes in charts inside eventing operator. Also changes in nats/jetstream would require chart updates.
+
+
+# Manifest operator
+Some components do not require any custom operator to install or upgrade (e.g. api-gateway, logging, monitoring) and use a base reconciler. With the operator approach, this task could be completed by a generic manifest operator. Custom resource for the manifest operator would contain information about chart location and overlay values. A single operator for multiple components can have some benefits in the in-cluster mode (better resource utilization) but would introduce challenges related to independent releases of charts and the manifest operator itself. Therefore a recommendation is that **all components will always provide the operator**. The manifest operator can be used as a template with a placeholder for your charts and default implementation (using the manifest library).
+
+# Component descriptor
+## OCM
+
+[OCM/CNUDIE](https://github.com/gardener/component-spec) stands for Open Component Descriptor and is used by Gardener. OCM intends to solve the problem of addressing, identifying, and accessing artefacts for software components, relative to an arbitrary component repository. By that, it also enables the transport of software components between component repositories. 
+
+## Operator bundle from Operator Lifecycle Manager (OLM)
+
+[Operator bundle](https://olm.operatorframework.io/docs/tasks/creating-operator-bundle/#operator-bundle) is a container image that stores Kubernetes manifests and metadata associated with an operator. A bundle is meant to represent a specific version of an operator on cluster.
+Operator bundle contains ClusterServiceVersion resource that describes operator version and installation descriptor:
+https://olm.operatorframework.io/docs/tasks/creating-operator-manifests/#writing-your-operator-manifests
+
+OLM provides also abstraction for operator catalog with release channels. What can solve the problem with modules discovery for users.
+
+## Own solution
+OCM is a simple descriptor and requires additional structures to apply it for installing, configuring and updating Kyma modules (operators). OLM addresses release channels and operator catalogs but we need to adjust it to the model with operators running centrally and remote subscriptions. We need to decide if we build our own solution or reuse existing software. 
+We need the representation of the component in the control plane cluster. The idea is to create own custom resource that can have OCM component descriptor embedded. The ModuleTemplate custom resource would contain:
+- release channel
+- CRD for the module (the one managed by module operator)
+- deployment of the operator (Kubernetes manifests / helm chart)
+
+
+# Component submission process
```

Comment:
Regarding github.com/kyma-project/community/blob/9f3f3cee5077af7c9f7b1ccca5c4d3fc2ae6e371/concepts/modularization/README.md#component-submission-process:

I would argue that we could not centrally plan  and scale efficiently for submissions of components. 

I think we could follow similar methods like Linux APT Registries or Krew Plugin Registries for kubectl:

The idea is to have a “registry” of “trusted authors/authorities” per control plane. By default we ship the “Kyma Authority” for Modules created by the Kyma Team. Any time a custom Developer wants to add a module, they will distribute the Authority Key (e.g. a Digest or a Public Key for Signature Verification) and we can either install that Authority Key by default (if its a well known module) or we let users install it manually.

We could embed this with OCM as OCM supports multiple signatures per module and would allow us to check for different trusted sources at once without changing the format of our component bundles